File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.java
import java.io.*;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        long Min = Long.parseLong(st.nextToken());
        long Max = Long.parseLong(st.nextToken());

        boolean[] check = new boolean[(int)(Max-Min+1)];
        for(long i=2;i*i<=Max;i++){
            long pow = i*i;
            long start = Min/pow;
            if(Min % pow!=0){
                start ++;
            }
            for(long j=start;j*pow<=Max;j++){
                check[(int)((j*pow)-Min)] = true;
            }
        }
        int cnt = 0;
        for(int i=0;i<=Max-Min;i++){
            if(!check[i]){
                cnt ++;
            }
        }
        System.out.println(cnt);
        
    }
}

File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.py
import sys
sys = sys.stdin.readline
import math

Min,Max = map(int,input().split())
check = [False]*(Max-Min+1)
for i in range(2,int(math.sqrt(Max))+1):
    pow = i*i #제곱수
    start = int(Min/pow)
    if Min%pow!=0:
        start +=1
    for j in range(start, int(Max/pow)+1):
        check[int((j*pow)-Min)] = True
cnt = 0
for i in range(0,Max-Min+1):
    if not check[i]:
        cnt +=1
print(cnt)